django_tenants/middleware/main.py

```diff
@@ -7,7 +7,10 @@
 from django.utils.deprecation import MiddlewareMixin
 
 from django_tenants.utils import remove_www, get_public_schema_name, get_tenant_types, \
-    has_multi_type_tenants, get_tenant_domain_model, get_public_schema_urlconf
+    has_multi_type_tenants, get_tenant_domain_model, get_public_schema_urlconf, get_tenant_model
+
+TENANTS_CACHE_DATA = {}
```

Comment:
Why not use the django.cache mechanism instead ? 

From what I can tell this may not be thread safe!